crates/dojo-core/src/utils/utils.cairo

```diff
@@ -0,0 +1,103 @@
+use core::option::Option;
+
+use dojo::model::{Layout, FieldLayout};
+
+/// Compute the poseidon hash of a serialized ByteArray
+pub fn hash(data: @ByteArray) -> felt252 {
+    let mut serialized = ArrayTrait::new();
+    core::serde::Serde::serialize(data, ref serialized);
+    core::poseidon::poseidon_hash_span(serialized.span())
+}
+
+/// Computes the entity id from the keys.
+///
+/// # Arguments
+///
+/// * `keys` - The keys of the entity.
+///
+/// # Returns
+///
+/// The entity id.
+pub fn entity_id_from_keys(keys: Span<felt252>) -> felt252 {
+    core::poseidon::poseidon_hash_span(keys)
+}
+
+/// TODO RBA
+pub fn find_field_layout(field_selector: felt252, field_layouts: Span<FieldLayout>) -> Option<Layout> {
+    let mut i = 0;
+    let layout = loop {
+        if i >= field_layouts.len() {
+            break Option::None;
+        }
+
+        let field_layout = *field_layouts.at(i);
+        if field_selector == field_layout.selector {
+            break Option::Some(field_layout.layout);
+        }
+
+        i += 1;
+    };
+
+    layout
+}
+
+/// Find the layout of a model field based on its selector.
+/// 
+/// # Arguments
+/// 
+/// * `model_layout` - The full model layout (must be a Layout::Struct).
+/// *  `member_selector` - The model field selector.
+/// 
+/// # Returns
+/// Some(Layout) if the field layout has been found, None otherwise. 
+pub fn find_model_field_layout(model_layout: Layout, member_selector: felt252) -> Option<Layout> {
+    match model_layout {
+        Layout::Struct(field_layouts) => {
+            find_field_layout(member_selector, field_layouts)
+        },
+        _ => {
+            // should never happen as model layouts are always struct layouts.
+            core::panic_with_felt252('Unexpected model layout');
+            Option::None
+        }
+    }
+}
+
+/// Indicates if at least of array item is None.
+pub fn any_none<T>(arr: @Array<Option<T>>) -> bool {
+    let mut i = 0;
+    let mut res = false;
+    loop {
+        if i >= arr.len() {
+            break;
+        }
+
+        if arr.at(i).is_none() {
+            res = true;
+            break;
+        }
+        i += 1;
+    };
+    res
+}
+
+/// Compute the sum of array items.
+/// Note that there is no overflow check as we expect
+/// small array items.
+pub fn sum(arr: Array<Option<u32>>) -> u32 {
```

Comment:
If the computation cost is the same with `u64` (as I guess it's felt base + some bits manipulation), we may switch to `u64`, `u128` or even `usize`?

Reply:
I will generalize the `sum` function 👍